examples/plot_power.py

```diff
@@ -1,86 +1,120 @@
-from params.power import fmin, fmax, power_method, is_epoched, freq_bands # noqa: E402
+###############################################################################
+# then set the parameters for PSD computation
 
+# filtering
+freq_bands = [[2, 4], [5, 7], [8, 12], [13, 29], [30, 59], [60, 90]]
+freq_band_names = ['delta', 'theta', 'alpha', 'beta', 'gamma1', 'gamma2']
 
-def create_infosource():
-    """Create node which passes input filenames to DataGrabber"""
+is_epoched = False
 
-    infosource = pe.Node(interface=IdentityInterface(fields=['subject_id',
-                                                             'sess_index']),
-                         name="infosource")
+fmin = 0.1
+fmax = 150
+sfreq = 600
+power_method = 'welch'  # for sensor PSD
+n_fft = 2048  # the FFT size (n_fft). Ideally a power of 2
 
-    infosource.iterables = [('subject_id', subject_ids),
-                            ('sess_index', sessions)]
+overlap = 0.5  # if is_epoched = False
 
-    return infosource
+###############################################################################
+# Then, we create our workflow and specify the `base_dir` which tells
+# nipype the directory in which to store the outputs.
 
+main_workflow = pe.Workflow(name='power_pipeline')
+main_workflow.base_dir = data_path
 
-def create_datasource():
+###############################################################################
+# Then we create a node to pass input filenames to DataGrabber from nipype
 
-    datasource = pe.Node(interface=nio.DataGrabber(infields=['subject_id',
-                                                             'sess_index'],
-                                                   outfields=['raw_file']),
-                         name='datasource')
+subject_ids = ['sub-0003']  # 'sub-0004', 'sub-0006'
+session_ids = ['ses-0001']
+infosource = create_iterator(['subject_id', 'session_id'],
+                             [subject_ids, session_ids])
 
-    datasource.inputs.base_directory = data_path
-    datasource.inputs.template = '*%s/%s/meg/%s*rest*ica.fif'
-    datasource.inputs.template_args = dict(raw_file=[['subject_id',
-                                                      'sess_index',
-                                                      'subject_id']])
-    datasource.inputs.sort_filelist = True
+###############################################################################
+# and a node to grab data. The template_args in this node iterate upon
+# the values in the infosource node
 
-    return datasource
+template_path = '*%s/%s/meg/%s*rest*ica.fif'
+template_args = [['subject_id', 'session_id', 'subject_id']]
+datasource = create_datagrabber(data_path, template_path, template_args)
 
+###############################################################################
+# Ephypype creates for us a pipeline which can be connected to these
+# nodes we created. To instantiate the preprocessing node, we import it
```

Comment:
preprocessing node? You mean power workflow?

Reply:
oh yes! I changed almost everywhere :-)